File: app/src/main/java/com/qiscus/sdksample/qiscussamplesdk/EmailAdapter.java
package com.qiscus.sdksample.qiscussamplesdk;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

/**
 * Created by zetra. on 9/20/16.
 */

public class EmailAdapter extends RecyclerView.Adapter<EmailAdapter.EmailVH> {

    private Context context;
    private List<String> emails;
    private OnClickListener clickListener;

    public EmailAdapter(Context context, List<String> emails) {
        this.context = context;
        this.emails = emails;
    }

    @Override
    public EmailVH onCreateViewHolder(ViewGroup parent, int viewType) {
        return new EmailVH(LayoutInflater.from(context).inflate(R.layout.item_email, parent, false), clickListener);
    }

    @Override
    public void onBindViewHolder(EmailVH holder, int position) {
        holder.bind(emails.get(position));
    }

    @Override
    public int getItemCount() {
        return emails.size();
    }

    public List<String> getEmails() {
        return emails;
    }

    public void setOnClickListener(OnClickListener clickListener) {
        this.clickListener = clickListener;
    }

    public class EmailVH extends RecyclerView.ViewHolder implements View.OnClickListener {

        private TextView emailText;
        private OnClickListener clickListener;

        public EmailVH(View itemView, OnClickListener clickListener) {
            super(itemView);
            emailText = (TextView) itemView.findViewById(R.id.email);
            this.clickListener = clickListener;

            itemView.setOnClickListener(this);
        }

        public void bind(String email) {
            emailText.setText(email);
        }

        @Override
        public void onClick(View v) {
            if (clickListener != null) {
                clickListener.onClick(getAdapterPosition());
            }
        }
    }

    public interface OnClickListener {
        void onClick(int position);
    }
}

File: app/src/main/java/com/qiscus/sdksample/qiscussamplesdk/EmailAdapter.kt
package com.qiscus.sdksample.qiscussamplesdk

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

/**
 * Created by zetra. on 9/20/16.
 */

class EmailAdapter(private val context: Context, val emails: MutableList<String>) : RecyclerView.Adapter<EmailAdapter.EmailVH>() {
    private var clickListener: OnClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EmailVH {
        return EmailVH(LayoutInflater.from(context).inflate(R.layout.item_email, parent, false), clickListener)
    }

    override fun onBindViewHolder(holder: EmailVH, position: Int) {
        holder.bind(emails[position])
    }

    override fun getItemCount(): Int {
        return emails.size
    }

    fun setOnClickListener(clickListener: OnClickListener) {
        this.clickListener = clickListener
    }

    inner class EmailVH(itemView: View, private val clickListener: OnClickListener?) :
            RecyclerView.ViewHolder(itemView), View.OnClickListener {

        private val emailText: TextView = itemView.findViewById(R.id.email) as TextView

        init {

            itemView.setOnClickListener(this)
        }

        fun bind(email: String) {
            emailText.text = email
        }

        override fun onClick(v: View) {
            clickListener?.onClick(adapterPosition)
        }
    }

    interface OnClickListener {
        fun onClick(position: Int)
    }
}
